### ISSUE TITLE
crash when connecting gql to hasura

### ISSUE THREAD

**Issue by mariusnita** [graphql-python/gql#92]
Hi, I'm trying this client with hasura and it crashes when executing the introspection query due to 
https://github.com/graphql-python/gql/blob/master/gql/transport/aiohttp.py#L109

If that line is just `operation_name` instead of `operation_name or ""` then it works. Hasura breaks when the operationName is set to `""`.

Not positive if this is a hasura issue or a gql issue.

**Comment by leszekhanusz** [comment]
Hi,

Could you please post here the complete error message ?

The PR #94 should fix your issue

**Comment by mariusnita** [comment]
Thanks!

The relevant part of the stack trace was this:
```
  File ".../lib/python3.8/site-packages/gql/client.py", line 88, in execute_async
    return await session.execute(document, *args, **kwargs)
  File ".../lib/python3.8/site-packages/gql/client.py", line 279, in execute
    await self.fetch_and_validate(document)
  File ".../lib/python3.8/site-packages/gql/client.py", line 247, in fetch_and_validate
    await self.fetch_schema()
  File ".../lib/python3.8/site-packages/gql/client.py", line 298, in fetch_schema
    self.client.schema = build_client_schema(self.client.introspection)
  File ".../lib/python3.8/site-packages/graphql/utilities/build_client_schema.py", line 56, in build_client_schema
    raise TypeError(
TypeError: Invalid or incomplete introspection result. Ensure that you are passing the 'data' attribute of an introspection response and no 'errors' were returned alongside: None.
```

And this was the hasura reply
```py
{'errors': [{'extensions': {'path': '$', 'code': 'validation-failed'}, 'message': 'no such operation found in the document: ""'}]}
```

**Comment by leszekhanusz** [comment]
Fixed in version v3.0.0a1
